plugin/completion.py

```diff
@@ -65,7 +65,14 @@ def format_completion(
         href = 'subl:lsp_run_text_command_helper {}'.format(args)
         details.append(make_link(href, 'More'))
     if lsp_label_detail and (lsp_label + lsp_label_detail).startswith(lsp_filter_text):
-        trigger = lsp_label + lsp_label_detail
+        if lsp_label_detail[0].isalnum() and lsp_label.startswith(lsp_filter_text):
```

Comment:
Is this `and lsp_label.startswith(lsp_filter_text)` necessary?

Reply:
If we want to keep the condition that filterText (if present) must be contained in the trigger, then it is necessary. For example if a server would set filterText to be exactly `label+labelDetails.detail`. But I don't think that it would happen in practice though. I've added it here, because I didn't want to change this condition how we treat filterText in this PR, but if we have another PR to be less restrictive about when or when not to choose filterText for the trigger, then it could be changed of course. Therefore I also did not add a test case for this which would enforce this condition.

If you thought about a possible performance impact, then I think that the `isalnum` condition already returns `False` in almost all cases, so it shouldn't matter for most servers.